src/email.rs

```diff
@@ -41,8 +38,8 @@ pub fn request(app: &Config, email_addr: EmailAddress, client_id: &str, nonce: &
 
     let session = session_id(&email_addr, client_id);
 
-    // Generate a 6-character one-time pad.
-    let chars: String = (0..6).map(|_| CODE_CHARS[rand::random::<usize>() % CODE_CHARS.len()]).collect();
+    // Generate a 8-character one-time pad.
+    let chars: String = (0..8).map(|_| CODE_CHARS[rand::random::<usize>() % CODE_CHARS.len()]).collect();
```

Comment:
I'd really like an actual calculation of entropy and a threat model to guide us, here. The OWASP page on [Insufficient Session-ID Length](https://www.owasp.org/index.php/Insufficient_Session-ID_Length) is a good example of a calculation that could give us reasonable assurance.